service/src/test/java/gov/nasa/pds/api/registry/model/PaginationLidvidBuilderTest.java

```diff
@@ -0,0 +1,273 @@
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"5", "6", "7", "8", "9"}, pageBuilder.page().toArray());
+    }
+
+    @Test
+    public void testComplexAddTrimHead3InsufficientData() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(7, 100));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"8", "9"}, pageBuilder.page().toArray());
+    }
+
+    @Test
+    public void testComplexAddTrimTail1() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(0, 8));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"1", "2", "3", "4", "5", "6", "7", "8"}, pageBuilder.page().toArray());
+    }
+
+    @Test
+    public void testComplexAddTrimTail2() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(0, 5));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"1", "2", "3", "4", "5"}, pageBuilder.page().toArray());
+    }
+
+    @Test
+    public void testComplexAddTrimTail3() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(0, 2));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"1", "2"}, pageBuilder.page().toArray());
+    }
+
+    @Test
+    public void testComplexAddTrimHeadAndTail1() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(1, 1));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"2"}, pageBuilder.page().toArray());
+    }
+
+
+    @Test
+    public void testComplexAddTrimHeadAndTail2() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(4, 1));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"5"}, pageBuilder.page().toArray());
+    }
+
+
+    @Test
+    public void testComplexAddTrimHeadAndTail3() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(7, 1));
+        for (List<String> input : complexInput) {
+            pageBuilder.addAll(input);
+        }
+        Assert.assertArrayEquals(new String[]{"8"}, pageBuilder.page().toArray());
+    }
+
+
+    @Test
+    public void testComplexAddTrimHeadAndTailSpanning() {
+        PaginationLidvidBuilder pageBuilder = new PaginationLidvidBuilder(new LidvidsContextStub(2, 5));
```

Comment:
Try any negatives like `LidvidsContextStub(2, -5)` or `LidvidsContextStub(-2, -5)`?

Reply:
Seemed defensible (albeit naive) to assume that a passed `LidvidsContext` would be valid.  If that validation isn't already baked into the existing `LidvidsContext` implementations I'll see what I can do there - _might_ be trivial but I need to verify that start/limit aren't used in any context where negatives are (say) expected, but handled after instantiation.

Plus I'll need to see how to throw in a way that will return a 422, which may be more trouble than is worth doing right this second (but I'll add it to the backlog).

Reply:
Ok to test raise an exception when limit or start is negative. Ok also to keep 422 error for later. 400 with a dedicated message sounds more than enough to me for now. Thanks